extensions/amp-accordion/0.1/amp-accordion.js

```diff
@@ -44,13 +51,20 @@ class AmpAccordion extends AMP.BaseElement {
   buildCallback() {
     this.sections_ = this.getRealChildren();
 
-    this.id_ = this.getSessionStorageKey_();
+    this.sessionId_ = this.getSessionStorageKey_();
+
+    if (isExperimentOn(this.win, TAG)) {
+      this.sessionOptOut_ = this.element.hasAttribute('disable-session-states');
+    }
 
     this.element.setAttribute('role', 'tablist');
 
-    // sessionStorage key: special created id for this element, this.id_.
+    // sessionStorage key: special created id for this element, this.sessionId_.
     // sessionStorage value: string that can convert to this.currentState_ obj.
-    this.currentState_ = this.getSessionState_();
+    if (!this.sessionOptOut_) {
```

Comment:
done.
Cons: we will generate sessionStorageKey for optout amp-accordion even it is not needed, Also I will keep record of current state during runtime even opt out. (but not write to session).
But the consolidation did make code cleaner, I suppose it is fine.